CODEOWNERS

```diff
@@ -0,0 +1,22 @@
+#
+# CODEOWNERS is a GitHub standard to specify who is automatically assigned pull requests to review. 
+# This helps to prevent pull requests from languishing without review.
+# GitHub can also be configured to require review from code owners before a pull request can be merged.
+#
+# Further reading is available from the following two URLs:
+#   https://blog.github.com/2017-07-06-introducing-code-owners/
+#   https://help.github.com/articles/about-codeowners/  
+#
+# It uses the same pattern rule for gitignore file
+# https://git-scm.com/docs/gitignore#_pattern_format
+#
+
+# Any pull request containing changes of files inside a 'data-plane' or 'client' directory will be reviewed 
+# by the azure-java-client-sdk-team.
+# Because there is a restructure coming up, and because 'client' is an overloaded term, there is a single
+# wildcard star being used, so we only match one directory, not many.
+*/data-plane/**       @azure/azure-java-client-sdk-team
+*/client/**           @azure/azure-java-client-sdk-team
+
+sdk/*/data-plane/**   @azure/azure-java-client-sdk-team
+sdk/*/client/**       @azure/azure-java-client-sdk-team
```

Comment:
I think @weshaggard is working through a new repo structure proposal that makes client go away.

Reply:
I see - he has it noted here: https://github.com/Azure/azure-sdk-for-java/issues/2847

It does indeed take away data-plane / client folders, and replaces it with '*-mgmt-*' and non-mgmt folders. I will close this PR and update it later once the structure of the CODEOWNERS is discussed more.